src/coreclr/jit/optimizer.cpp

```diff
@@ -3166,6 +3166,9 @@ bool Compiler::optCanonicalizeExit(FlowGraphNaturalLoop* loop, BasicBlock* exit)
     }
 
     optSetWeightForPreheaderOrExit(loop, newExit);
+
+    JITDUMP("Created new exit " FMT_BB " to replace " FMT_BB " exit for " FMT_LP "\n", newExit->bbNum, exit->bbNum,
+            loop->GetIndex());
```

Comment:
`optSetWeightForPreheaderOrExit` references the block created, so moving its dumping here makes the jitdump look a bit strange I think.

Reply:
I added the "Canonicalize exit" line above so it's obvious what process is starting when new blocks start getting created -- before I was confused at why new blocks were getting created. So moving this line to the end "brackets" the process by telling you what happened when it's done. E.g.,
```
Canonicalize exit BB15 for L00 to have only loop predecessors
New Basic Block BB18 [0116] created.
Setting edge weights for BB18 -> BB15 to [0 .. 3.402823e+38]
setting likelihood of BB18 -> BB15 to 1
  Estimated likelihood BB04 -> BB18 to be 0.6585233 (contribution: 35.62623)
Setting edge weights for BB04 -> BB18 to [35.62623 .. 35.62623]
  Estimated likelihood BB05 -> BB18 to be 0.3993941 (contribution: 10.80365)
Setting edge weights for BB05 -> BB18 to [10.80365 .. 10.80365]
Setting edge weights for BB18 -> BB15 to [46.42988 .. 46.42988]
Created new exit BB18 to replace BB15 exit for L00
```